app/src/main/java/org/thoughtcrime/securesms/home/HomeActivity.kt

```diff
@@ -309,19 +352,92 @@ class HomeActivity : PassphraseRequiredActionBarActivity(),
             }
-            configFactory.user?.let { user ->
-                if (!user.isBlockCommunityMessageRequestsSet()) {
-                    user.setCommunityMessageRequests(false)
+            configFactory.user
+                ?.takeUnless { it.isBlockCommunityMessageRequestsSet() }
+                ?.setCommunityMessageRequests(false)
+        }
+    }
+
+    @Preview
+    @Composable
+    fun PreviewMessageDetails(
+        @PreviewParameter(ThemeResPreviewParameterProvider::class) themeResId: Int
+    ) {
+        PreviewTheme(themeResId) {
+            SeedReminder()
+        }
+    }
+
+    @Composable
+    private fun SeedReminder() {
+        AppTheme {
+            Column {
+                Box(
+                    Modifier
+                        .fillMaxWidth()
+                        .height(4.dp)
+                        .background(MaterialTheme.colors.secondary))
+                Row(
+                    Modifier
+                        .background(MaterialTheme.colors.surface)
+                        .padding(horizontal = 24.dp, vertical = 16.dp)
+                ) {
+                    Column(Modifier.weight(1f)) {
+                        Row {
+                            Text(stringResource(R.string.save_your_recovery_password), style = MaterialTheme.typography.h8)
+                            Spacer(Modifier.requiredWidth(8.dp))
+                            SessionShieldIcon()
+                        }
+                        Text(stringResource(R.string.save_your_recovery_password_to_make_sure_you_don_t_lose_access_to_your_account), style = MaterialTheme.typography.small)
+                    }
+                    Spacer(Modifier.width(12.dp))
+                    OutlineButton(
+                        textId = R.string.continue_2,
+                        modifier = Modifier
+                            .align(Alignment.CenterVertically)
+                            .contentDescription(R.string.AccessibilityId_reveal_recovery_phrase_button),
+                        onClick = { start<RecoveryPasswordActivity>() }
+                    )
                 }
             }
         }
     }
 
+    @Composable
+    private fun EmptyView(newAccount: Boolean) {
+        AppTheme {
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                modifier = Modifier
+                    .padding(horizontal = 50.dp)
+                    .padding(bottom = 12.dp)
+            ) {
+                Spacer(modifier = Modifier.weight(1f))
+                Icon(
+                    painter = painterResource(id = if (newAccount) R.drawable.emoji_tada_large else R.drawable.ic_logo_large),
+                    contentDescription = null,
+                    tint = Color.Unspecified
+                )
+                if (newAccount) Text(stringResource(R.string.onboardingAccountCreated), style = MaterialTheme.typography.h4, textAlign = TextAlign.Center)
```

Comment:
Any reason the if is repeated twice? Shouldn't they both be in the one if block?